### PR TITLE
fees: Reduce minimum relay fee to 0.001 DOGE

### PR DESCRIPTION
**Pull request by michilumin**
This loosens the minimum set of fee values for 1.14.4 to enable the rest in 1.14.5. Please review the figures, they should match the proposal, if not do let us know. If any values need to change for 1.14.4 and have been missed, equally let us know. Otherwise we'll raise the second set as a follow-on PR.

### CODE REVIEW

File: qa/rpc-tests/bumpfee.py
Comment (rnicoll) [inline qa/rpc-tests/bumpfee.py:61]: Quick explanatory note on this test move; the outputs picked by the node depend on the test ordering (ugh), and at the later stages the output being picked consistently did not have enough change for a fee bump to succeed.

**Review by patricklodder — DISMISSED** [review]
Needs some work:

1. By not setting recommended minimum inclusion limits for miners, this PR would completely disable spam management and pushes miners to change their configuration (opt-in spam management rather than opt-out, which is a huge step, as we're coming from network-wide developer enforced fees) This can be resolved by setting `DEFAULT_BLOCK_MIN_TX_FEE` to `COIN / 100`.
2. All changes in wallet.h need to be undone, because those will lead to integration issues with past clients. Can simply reset that entire file.
3. Needs a rebase after we merge #2439 because this contains some changes from there.
4. Per [contribution guidelines](https://github.com/dogecoin/dogecoin/blob/1.14.4-dev/CONTRIBUTING.md), this could REALLY benefit from integration tests that make sure that:
    - 1.14.4 wallets still are able to send transactions that get mined by a 1.14.3-style parametrized network
    - 1.14.3 wallets still are able to send transactions that get mined by a 1.14.4-default configured network
    - planned changes for 1.14.5 can be mined by a 1.14.4 network
    - these tests can be fully parametrized with the recent changes to configurability.

**Comment by patricklodder** [comment]
Apart from review - I noticed that item 4 from the fee plan was implemented differently (wallet only.) What is the rationale for that?

File: src/primitives/transaction.h
Comment (rnicoll) [reply src/primitives/transaction.h:16]: My fault, it's bled in from a draft branch I had before you raised your dust limit parameter. I'll raise a PR to revert it on Michi's branch.

**Comment by rnicoll** [comment]
> 
> 
> Apart from review - I noticed that item 4 from the fee plan was implemented differently (wallet only.) What is the rationale for that?

Item 4 being free transaction space, right? It will mean we missed somewhere to change it, but if we're reverting wallet.h, can tackle on the next iteration.

**Comment by rnicoll** [comment]
If I've understood the ask on integration tests, they're actually more tests of the policy, is that correct? As in, we're not testing the currently set value against previous versions (because that test would break on expected changes in 1.14.5), but instead simulating nodes with those values?

If so, I think it can be raised as a separate PR, and I'll do that.

**Comment by patricklodder** [comment]
> Item 4 being free transaction space, right? It will mean we missed somewhere to change it, but if we're reverting wallet.h, can tackle on the next iteration.

Right. But there's something... just was curious why I didn't see the implementation. The problem that I see with item 4 right now is that there is a conflict between fee filter and free transaction space. There are notes authored by you in the current qa testset, originating from #1431, commenting on unavailability of free transaction space that may be the cause of this, but I do not have a clear view of what was done to cause that, yet, and I suspect that we need a network protocol feature to properly manage free transaction relay, and then bump the version, to resolve this for good. So I already hinted that we may need to do the free transaction space with 1.14.5 instead of 1.14.4, so that we can do that protocol extension properly. This would be an acceptable outcome, because I think it's more important to have properly functioning features than saying it's in and then there being hidden caveats, as was done with 1.14.2.

As for the wallet.h change. Any wallet changes will conflict with the existing network.

> If I've understood the ask on integration tests, they're actually more tests of the policy, is that correct?

Incorrect. The policy elements are:

1. Get rid of the 1.14.2 hardcoded fees and do a true recommendation-only that can be changed by any individual node operator, without needing to compile new software.
2. Do an updated recommendation that is effectively 100x lower than the enforced hardcoded fee is today.

I had [shared a WIP test with you over the weekend](https://github.com/patricklodder/dogecoin/blob/1.14.4-mintxfee-tests/qa/rpc-tests/paytxfee.py), building on #2437, that, on top of the [test introduced with the dust limit configuration](https://github.com/dogecoin/dogecoin/blob/1.14.4-dev/qa/rpc-tests/harddustlimit.py), tests and ensures element 1. The success of element 2 can really only be measured after the recommendation has been done and wallet software - including Dogecoin Core - implement it and we have some results on mainnet.

1.14.4 implements prerequisites that enable recommending a **_lower_** fee than currently hardcoded since 1.14.2. On mainnet, transition take time, as people need to update their live nodes, but on regtest we can perfectly test (now that everything is configurable) whether the changes we do now enable the updated recommendation and have no adverse effects for the network as a whole, or node operators upgrading from 1.14.2 or 1.14.3 to 1.14.4. This means we can test integration against a simulated 1.14.3 network right now for the current planned release, and the effects that a new wallet will have once this has been adopted on mainnet. And since we can, we should, because that proves that things work rather than a "Core Developer" claiming that it works.

**Comment by etherx-dev** [comment]
Good. Now remove the hardcoded limitations completely for us legacy users.

**Comment by patricklodder** [comment]
> Good. Now remove the hardcoded limitations completely for us legacy users.

Already done with #2368, #2348, #2437 and will be released with 1.14.4, which we are finalizing right now - see [my WIP fee test](https://github.com/patricklodder/dogecoin/blob/b085104b5ea799d1f0dfe6c178a1fb01ca7f0fcd/qa/rpc-tests/dogecoin-fees.py) for proof - this already works on current 1.14.4-dev branch.

**Review by langerhans — APPROVED** [review]
No objections from my side

**Pull request merged**
